### PR TITLE
[Merged by Bors] - chore(Topology/{Compactness/Compact}, Irreducible}): rename type variables

### PR DESCRIPTION
**Pull request by grunweg**
X, Y, Z are standard mathematical names for topological spaces.
As [discussed](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Naming.20convention.3A.20topological.20spaces/near/395769893) on zulip, let us rename them.

As a drive-by commit, re-use the declared variables $\iota$, $s$ and $t$ (more) when stating theorems.

--------

I'm following by checklist [here](https://github.com/leanprover-community/mathlib4/pull/7589#issuecomment-1755212582).

### CODE REVIEW

**Comment by eric-wieser** [comment]
> X, Y, Z are standard mathematical names for topological spaces.

I'm not sure if this is true in mathlib; I've seen greek letters in lots of places too. There is only one mention of topology in the style guide, and it uses `[DiscreteTopology α]`!

File: Mathlib/Topology/Algebra/Group/Basic.lean [hunk 1]
```diff
@@ -1295,7 +1295,7 @@ theorem MulAction.isClosedMap_quotient [CompactSpace α] :
     IsClosedMap (Quotient.mk' : β → Quotient (orbitRel α β)) := by
   intro t ht
   rw [← quotientMap_quotient_mk'.isClosed_preimage, MulAction.quotient_preimage_image_eq_union_mul]
-  convert ht.smul_left_of_isCompact (isCompact_univ (α := α))
+  convert ht.smul_left_of_isCompact (isCompact_univ (X := α))
```
Comment (eric-wieser) [inline Mathlib/Topology/Algebra/Group/Basic.lean:1298]: This is possibly evidence that the `α` name was better

Comment (grunweg) [reply Mathlib/Topology/Algebra/Group/Basic.lean:1298]: I agree this demonstrates an inconsistency. My explanation is that the section on line 1260 onwards uses Greek letters. (I'd be happy to change that also. I decided to start with one file.)

**Comment by grunweg** [comment]
> > X, Y, Z are standard mathematical names for topological spaces.
> 
> I'm not sure if this is true in mathlib; I've seen greek letters in lots of places too. There is only one mention of topology in the style guide, and it uses `[DiscreteTopology α]`!

I agree that Greek letters are persive in topology files. In math textbooks, I've always seen X, Y, Z etc., though. Hence, my guess was that general topology files, being old enough, were written when Greek letter prevailed. (It's possible I'm wrong, though!)

**Comment by grunweg** [comment]
I've [asked on zulip](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Naming.20convention.3A.20topological.20spaces).

File: Mathlib/Topology/Algebra/Group/Basic.lean (hunk 1 above)
Comment (grunweg) [reply Mathlib/Topology/Algebra/Group/Basic.lean:1298]: For zulip, it seems going ahead with this is wanted. Shall I change this section in this PR also?

**Comment by grunweg** [comment]
awaiting-author

**Comment by PatrickMassot** [comment]
I'm sorry we haven't been quick enough to merge that one. Could you please try to fix the conflicts?

**Comment by grunweg** [comment]
@PatrickMassot Updated; as soon as this passes CI, it should be ready to review.
The merge meant I had to basically re-do these changes; it might be easier to just review/skim all commits since the merge separately. (They're logically independent.)

File: Mathlib/Topology/Compactness/Compact.lean [hunk 2]
```diff
@@ -26,49 +26,51 @@ We define the following properties for sets in a topological space:
 -/
 open Set Filter Topology TopologicalSpace Classical
 
+set_option autoImplicit true
```
Comment (sgouezel) [inline Mathlib/Topology/Compactness/Compact.lean:29]:
```suggestion
set_option autoImplicit true
```
Please don't do that. It just makes things more unpredictable and harder to read.

**Comment by grunweg** [comment]
awaiting-review

**Comment by grunweg** [comment]
Thanks for the quick reviews, this time!

**Pull request closed without merging**